From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Outfit8TSB <sunnybunnies09@naver.com>
Date: Tue, 23 Apr 2021 01:27:00 +0900
Subject: [PATCH] Fix Adventure

diff --git a/build.gradle.kts b/build.gradle.kts
index 65e67b87..38d2c251 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -26,6 +26,10 @@ configurations.api {
     extendsFrom(apiAndDocs)
 }

+repositories {
+    maven("https://github.com/GAME-CLI-SRV-DEV/NogyangSpigot-Terminals-Repos/tree/main/snapshots/")
+    }
+
 dependencies {
     // api dependencies are listed transitively to API consumers
     api("com.google.guava:guava:32.1.2-jre")

